### DISCUSSION TITLE
Iobroker integration

### DISCUSSION THREAD

**Discussion by mitro89** [BlazejosP/huawei-sun2000-API-CLI#10] [Ideas]
Hello, is it also possible to get the data with that solution in the iobroker, if this is running on a raspberry?

**Comment by BlazejosP** [comment]
Hello and thanks for that idea!
To be honest I first time herd about this software personally I myself  use domoticz+mqtt+influxDB/Grafana
You can found this if you look inside code in kioskmode.sh  I prepared as for now export to this software 
+influxDB->Grafana
+Domoticz
+MQTT

Why because i have this software and I tested this implementations and they work.
Now I'm on the stage to implement this also to second file fusionsolarapp.sh
And if you are interested I can also add export to ioBroker but need some time to write this function myself around a month. because must install this software and made some investigations how I can  use bash in this software. Solution probably will be   write function which will use command line tool for sending data to ioBroker like in case of MQTT. There is also quick&dirty workaround you can use now even today sending data to MQTT server in kioskmode.sh and ioBroker has MQTT client as I found in ioBroker webpage that may work  and be done quitefast  but for now only in Kioskmode API. 
If you d'like write bash function yourself I will be really happy and I will add you as contributor to this repository and include your contribution in the credits. To summarize there are three solutions.

1) fast quick&dirty use  MQTT sending option from kioskmode.sh  then configure IoBroker with MQTT client and you have data from this software inside IoBroker (minus is MQTT now works  only now with kiskomede API so only basic data but in not so long time I add also MQTT to fusionsolar.sh) but you can prepare this in one two days just configure your installation with MQTT server and IoBroker plugin MQTT client
2)  If you write function in bash which will deal with IoBroker sending I happily integrate this inside code (if you d'like do this I try help you as much as I can you have this software already configured so probably will be fast or slow  depending on your skills) 
3) Wait until I myself install some test installation of IoBroker and start writing code. As for now I d'like finish fusionsolarapp.sh and integrate inside sending to influxDB,Domoticz,MQTT  and implement one config file so realistically I can start integration of 
 ioBroker on beginning of April when my previous task will be done.

Write me which solution will be most suitable for you?
